### DISCUSSION TITLE
Telegram Speech to Text bot on Pi 4

### DISCUSSION THREAD

**Discussion by tonym128** [ggml-org/whisper.cpp#218] [Show and tell]
I wrote a Telegram Bot to transcribe your speech to text using whisper.cpp. For me, this is for when someone sends me a voice note that I can't listen to straight away and sometimes don't listen to for hours on end because of being in meetings or some other reason. It accepts voice notes, short videos, forwarded messages from WhatsApp. There's other services that can do this, and it's not fast, but it's free, whereas the alternatives I have found are not.

Thanks to Whisper.cpp it's running permanently (for now) on a PI 4 2gb using the base.en model. 

Give it a voice note try 😄 

[@shhhhhhhhhhhhhhhhh_bot](https://t.me/shhhhhhhhhhhhhhhhh_bot
) waiting for your message

**Comment by elvisi01** [comment]
Nice! Do you have it explained this somewhere? I'm trying to do just that on a Raspberry pi 3b, but I don´'t have the requiered knowledge... I tried anothe project called [WhisperAI_Tg](https://github.com/HeenLight/WhisperAI_Tg), bu I cannot make it work.

Thanks in advance!

**Reply by tonym128** [reply]
Sorry I missed this at the time.

I used this version of Whipser and it worked on my Pi4 https://github.com/ggerganov/whisper.cpp It should work on the Pi 3B with the smaller models too.
